src/coreclr/tools/Common/Compiler/InstructionSetSupport.cs

```diff
@@ -64,38 +64,32 @@ public bool IsInstructionSetExplicitlyUnsupported(InstructionSet instructionSet)
 
         public InstructionSetSupportFlags Flags => _flags;
 
-        public static string GetHardwareIntrinsicId(TargetArchitecture architecture, TypeDesc potentialTypeDesc)
+        public static string GetHardwareIntrinsicId(TargetArchitecture architecture, TypeDesc potentialTypeDesc, out bool unsupportedSubset)
         {
+            unsupportedSubset = false;
+
             if (!potentialTypeDesc.IsIntrinsic || !(potentialTypeDesc is MetadataType potentialType))
                 return "";
 
-            if (architecture == TargetArchitecture.X64)
-            {
-                if (potentialType.Name == "X64")
-                    potentialType = (MetadataType)potentialType.ContainingType;
-                if (potentialType.Name == "VL")
-                    potentialType = (MetadataType)potentialType.ContainingType;
-                if (potentialType.Namespace != "System.Runtime.Intrinsics.X86")
-                    return "";
-            }
-            else if (architecture == TargetArchitecture.X86)
+            if (architecture is TargetArchitecture.X64 or TargetArchitecture.X86)
             {
                 if (potentialType.Name == "X64")
+                {
                     potentialType = (MetadataType)potentialType.ContainingType;
+                    unsupportedSubset = architecture == TargetArchitecture.X86;
+                }
                 if (potentialType.Name == "VL")
                     potentialType = (MetadataType)potentialType.ContainingType;
                 if (potentialType.Namespace != "System.Runtime.Intrinsics.X86")
                     return "";
             }
-            else if (architecture == TargetArchitecture.ARM64)
+            else if (architecture is TargetArchitecture.ARM64 or TargetArchitecture.ARM)
```

Comment:
There is no intrinsic support for `TargetArchitecture.ARM` today (by any of our runtimes). Does this need to report the type or can it accordingly just report it as "unsupported" for the time being?

Reply:
I folded it to follow the same logic as X64/X86. The ARM logic existed there before the change.